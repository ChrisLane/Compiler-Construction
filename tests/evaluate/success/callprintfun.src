// This is to test the evaluation of calling a function with arguements that are then used in the arguement. This should output 0, 10, 100, 3, 0 on new lines.
print123() { print $a; print $b; print 3; }

main() {
    print 0;
    var a = 10;
    var b = $a * $a;
    print123(a, b);
    print 0;
}
