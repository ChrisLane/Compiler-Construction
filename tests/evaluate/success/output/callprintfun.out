0
10
100
3
0
  
